src/Features/Core/Portable/Completion/Providers/AbstractTypeImportCompletionProvider.cs

```diff
@@ -223,14 +232,32 @@ internal override async Task<CompletionChange> GetChangeAsync(Document document,
 
                 return CompletionChange.Create(Utilities.Collapse(newText, builder.ToImmutableAndFree()));
             }
-            else
+
+            static bool ShouldCompleteWithFullyQualifyTypeName(Document document)
             {
-                // For workspace that doesn't support document change, e.g. DebuggerIntellisense
-                // we complete the type name in its fully qualified form instead.
-                var fullyQualifiedName = $"{containingNamespace}.{completionItem.DisplayText}";
-                var change = new TextChange(completionListSpan, fullyQualifiedName);
+                var workspace = document.Project.Solution.Workspace;
 
-                return CompletionChange.Create(change);
+                // Certain types of workspace don't support document change, e.g. DebuggerIntellisense
+                if (!workspace.CanApplyChange(ApplyChangesKind.ChangeDocument))
```

Comment:
I think you can use feature service from debugger workspace to disable it rather than using CanApplyChange just for debugger workspace?

Reply:
Let me try that

Reply:
Nope, doesn't work. `workspace.Services.GetService<IDocumentSupportsFeatureService>().SupportsRefactorings(document)` returns true for `DebuggerIntellisenseWorkspace`

Reply:
sure, you need to export one for DebuggerIntellisenseWorkspace that return false? IDocumentSupportsFeatureService is per workpsace service which supposed to return different value based on workspace? like we have one for interactive and etc?